WcaOnRails/app/models/registration.rb

```diff
@@ -140,7 +140,7 @@ def waiting_list_info
   def events_to_associated_events(events)
     events.map do |event|
       competition_event = competition.competition_events.find_by!(event: event)
-      registration_competition_events.find_by_competition_event_id(event.id) || registration_competition_events.build(competition_event: competition_event)
+      registration_competition_events.find_by_competition_event_id(competition_event.id) || registration_competition_events.build(competition_event: competition_event)
```

Comment:
Just to clarify: this mistaken variable name caused the *find* to fail (return `nil`) so that we built a new object.